web_client/src/scenes/SouffleStation/components/SouffleButton.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled, { css } from 'react-emotion';
+import ReactCSSTransitionReplace from 'react-css-transition-replace';
+import colors, { artistEmoji, albumEmoji } from 'theme';
+import { CtaButton } from 'components/buttons';
+import BackForthIcon from './BackForthSvg';
+
+const emojiBySouffleType = {
+  artist: artistEmoji,
+  album: albumEmoji
+};
+
+/**
+ *  Crossfade css transition classes for a ReactCSSTransition.
+ */
+const crossFade = css`
+  &-leave {
+    opacity: 1;
+  }
+  &-leave&-leave-active {
+    opacity: 0;
+    transition: opacity .5s ease-out;
+  }
+  &-enter {
+    opacity: 0;
+  }
+  &-enter&-enter-active {
+    opacity: 1;
+    transition: opacity .5s ease-out;
+  }
+`;
+
+/**
+ *  This is a very annoying way to get the artist / album emoji to crossfade when changed.
+ */
+const souffleButtonText = (isResouffle, souffleBy) => {
+  const emoji = emojiBySouffleType[souffleBy];
+  const text = (isResouffle ? 'ReSouffle' : 'Souffle') + ' by ' + emoji;
+  return (
+      <ReactCSSTransitionReplace
```

Comment:
This ended up super messy because I was initially just trying to crossfade the emojis, but the only way I was able to get it sort of working is to crossfade the entire line of text.

When I get the chance, I will (at least):
- Make `souffleButtonText` what it originally was: a function that takes `isResouffle` and `souffleBy` and returns the button text.
- Move this `ReactCSSTransitionReplace` into the `SouffleButton` render.